deploy/role.yaml

```diff
@@ -29,6 +29,7 @@ rules:
   - monitoring.coreos.com
   resources:
   - servicemonitors
+  - prometheusrules
```

Comment:
Do we support deletion of prometheus objects ?

Reply:
For when sno is redeployed with prometheus disabled (e.g in helm chart)

Reply:
good point. we have to support object deletion as well. I handle it in 
- https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/769
but it's better to merge that changes into these PR.